module/PowerShellEditorServices/Start-EditorServices.ps1

```diff
@@ -173,8 +176,7 @@ function New-NamedPipeName {
 
     # We try 10 times to find a valid pipe name
     for ($i = 0; $i -lt 10; $i++) {
-        # add a guid to make the pipe unique
-        $PipeName = "PSES_$([guid]::NewGuid())"
+        $PipeName = "PSES_$([System.IO.Path]::GetRandomFileName())"
```

Comment:
It makes pipe name length shorter.

Clarifications:
At system-level the pathname of a Unix domain socket is limited in length and some libraries also have such a restriction, see *Maximum path length* section on [junixsocket](https://code.google.com/archive/p/junixsocket/wikis/GettingStarted.wiki) that I use.

Because as it is now, the pipe name is really long esp. on Linux where it is the Temp folder (usually such as "/var/folders/92/2yqlfyrj3lqcvrhtdwxj8djc0000gp/T/") + long enough guid as a pipe name. The *junixsocket* library will throw an exception with the use of `[guid]::NewGuid`.

Btw Unix sockets (which named pipes are represented by on Linux) recommended path location is /var/run/ (see */var/run : Run-time variable data* section here: http://www.pathname.com/fhs/pub/fhs-2.3.html but std library creates pipe in a temp directory. So it might make sense to reconsider the directory on Linux.

Reply:
> So it might make sense to reconsider the directory on Linux.

I'm with you, but IIRC I don't think we have this constructor available to us in .NET Core 2.0 (which is what PowerShell Core 6.0 uses) that lets us pass in a path name of where we want the pipe (or unix socket) to be saved 😢

Reply:
p.s. love the use of `GetRandomFileName`!